### PR TITLE
Add optional id parameter for openInPortal

### PR DESCRIPTION
**Pull request by nturinski**
https://github.com/Microsoft/vscode-azureappservice/issues/311

Portal changed the path for individual slots to include the parent's path anymore

### CODE REVIEW

**Comment by nturinski** [comment]
The optional id allows two things:

1.) Opening a resource in Azure that does not necessarily live in the explorer
2.) Opening a resource in Azure where the id does not match the conventional parent/child id pattern

File: ui/src/treeDataProvider/AzureNode.ts [hunk 1]
```diff
@@ -88,8 +88,10 @@ export class AzureNode<T extends IAzureTreeItem = IAzureTreeItem> implements IAz
         await this.treeDataProvider.refresh(this);
     }
 
-    public openInPortal(): void {
-        (<(s: string) => void>opn)(`${this.environment.portalUrl}/${this.tenantId}/#resource${this.id}`);
+    public openInPortal(id?: string): void {
+        id ?
```
Comment (EricJizbaMSFT) [inline ui/src/treeDataProvider/AzureNode.ts]: You can simplify by doing one of the following:
`id = id === undefined ? this.id : id`
`id = id || this.id`

That way you don't have duplicate the open and url logic

Comment (nturinski) [reply ui/src/treeDataProvider/AzureNode.ts]: This is why you are "boss".

File: ui/src/treeDataProvider/AzureNode.ts [hunk 2]
```diff
@@ -88,8 +88,9 @@ export class AzureNode<T extends IAzureTreeItem = IAzureTreeItem> implements IAz
         await this.treeDataProvider.refresh(this);
     }
 
-    public openInPortal(): void {
-        (<(s: string) => void>opn)(`${this.environment.portalUrl}/${this.tenantId}/#resource${this.id}`);
+    public openInPortal(id?: string): void {
+        id = id === undefined ? id : this.id;
+        (<(s: string) => void>opn)(`${this.environment.portalUrl}/${this.tenantId}/#resource/${id}`);
```
Comment (EricJizbaMSFT) [inline ui/src/treeDataProvider/AzureNode.ts]: Did you intentionally add a slash?:
old: `resource${this.id}`
new: `resource/${id}`

Comment (nturinski) [reply ui/src/treeDataProvider/AzureNode.ts]: No, that was an accident from when I was calling opn twice.  Good catch.

File: ui/src/treeDataProvider/AzureNode.ts [hunk 3]
```diff
@@ -88,8 +88,9 @@ export class AzureNode<T extends IAzureTreeItem = IAzureTreeItem> implements IAz
         await this.treeDataProvider.refresh(this);
     }
 
-    public openInPortal(): void {
-        (<(s: string) => void>opn)(`${this.environment.portalUrl}/${this.tenantId}/#resource${this.id}`);
+    public openInPortal(id?: string): void {
+        id = id === undefined ? id : this.id;
```
Comment (EricJizbaMSFT) [inline ui/src/treeDataProvider/AzureNode.ts]: Your conditionals are backwards

**Pull request merged**